File: test/csharp/SpecGate.CtscFixtures/Conformance/Structured/ScalarOperators.cs
using SpecGate.Annotations;

namespace SpecGate.CtscFixtures.Conformance.Structured;

[SpecEvent("Measurement")]
public sealed class Measurement
{
    [SpecEvent("temperature")]
    public int Temperature { get; set; }

    [SpecEvent("label")]
    public string Label { get; set; } = string.Empty;

    [SpecEvent("readings")]
    public List<int> Readings { get; set; } = [];
}

public static class ScalarOperators
{
    [SpecOperation("get_measurement", Spec = "fixture.scalar_operators")]
    public static Measurement GetMeasurement() =>
        new()
        {
            Temperature = 72,
            Label = "sensor-A3-north",
            Readings = [68, 70, 72, 71, 73],
        };

    [SpecOperation("get_empty", Spec = "fixture.scalar_operators")]
    public static List<string> GetEmpty() => [];
}

File: test/rust/crates/specgate-ctsc-fixtures/src/conformance/structured/scalar_operators.rs
use specgate::{SpecEvent, spec_operation};

#[derive(Debug, Clone, PartialEq, Eq, SpecEvent)]
#[spec_component("fixture.scalar_operators")]
pub struct Measurement {
    #[spec_event]
    pub temperature: i32,
    #[spec_event]
    pub label: String,
    #[spec_event]
    pub readings: Vec<i32>,
}

#[spec_operation("get_measurement", spec = "fixture.scalar_operators")]
pub fn get_measurement() -> Measurement {
    Measurement {
        temperature: 72,
        label: "sensor-A3-north".to_string(),
        readings: vec![68, 70, 72, 71, 73],
    }
}

#[spec_operation("get_empty", spec = "fixture.scalar_operators")]
pub fn get_empty() -> Vec<String> {
    Vec::new()
}

#[test]
fn scalar_values_are_asserted_without_matcher_dsl() {
    let measurement = get_measurement();
    assert!(measurement.temperature > 70);
    assert!(measurement.label.starts_with("sensor-"));
    assert_eq!(measurement.readings.len(), 5);
    assert!(get_empty().is_empty());
}
